config/nrfconnect/chip-module/Kconfig.features

```diff
@@ -73,6 +73,21 @@ config SPI_NOR_FLASH_LAYOUT_PAGE_SIZE
 
 endif # CHIP_SPI_NOR
 
+config CHIP_MEMORY_PROFILING
+	bool "Enable features for tracking memory usage"
+	default n
+	select CHIP_STATISTICS
+	select CHIP_MALLOC_SYS_HEAP_WATERMARKS_SUPPORT
+	select MBEDTLS_MEMORY_DEBUG if !CHIP_CRYPTO_PSA
+	select SYS_HEAP_RUNTIME_STATS
+	select KERNEL_SHELL
```

Comment:
This is needed to get stack usage, right? Then don't we need other options like: `CONFIG_THREAD_MONITOR`, `CONFIG_STACK_INFO` and `CONFIG_INIT_STACK`?

Reply:
It seems that it is already enabled, I need to investigate why because I agree that it shouldn't be :)

Reply:
Ok, it seems that enabling KERNEL_SHELL `implies` enabling each of the mentioned configs.

Reply:
ok, thanks!